mercantile/__init__.py

```diff
@@ -494,18 +507,19 @@ def parent(*tile, **kwargs):
     """
     tile = _parse_tile_arg(*tile)
 
+    x, y, z = tile
+
+    if z == 0:
+        raise ParentTileError("root tile does not have a parent")
+
     # zoom is a keyword-only argument.
     zoom = kwargs.get("zoom", None)
 
-    if zoom is not None and (tile[2] < zoom or zoom != int(zoom)):
+    if zoom is not None and (z <= zoom or zoom != int(zoom)):
```

Comment:
This is an edge case we haven't handled before: parent(t, z) with t.z == z makes no sense; the parent of a tile on the tile's zoom level that is.